index.ts

```diff
@@ -68,103 +64,132 @@ const server = new Server(
 interface ToolDefinition {
   tool: Tool;
   handler: (args: any) => Promise<any>;
+  requiresAuth?: boolean;
```

Comment:
this is just a nitpick: since this is most commonly true, it forces us to almost always add this to every tool. I'd chose a name where the most common use case is false/undefined as default. for example: `worksAsGuest`, or `noAuthRequired`.